src/js/datepicker.js

```diff
@@ -84,71 +49,52 @@ DatePicker = class {
     buttonNext.textContent = "Next";
     datepickerNav.append(buttonNext);
 
-    let monthView = new CalendarMonthView(
-      this.currentMonth.monthName,
-      this.currentMonth.firstWeekdayOfMonth,
-      this.currentMonth.numberOfDays,
-      this.currentMonth.year,
-      this.currentMonth.isCurrentMonth,
-      this.currentMonth.dayNumToday
-    );
-    calendarMonthWrap.appendChild(monthView.monthView);
 
     calendarHtml.appendChild(datepickerNav);
     calendarHtml.appendChild(calendarMonthWrap);
 
     // not sure why I can't use this.currentMonthIndex directly here, comes back as undefined
     let currentMonthIndex = this.currentMonthIndex;
-    let months = this.months;
-    let yearInView = this.currentYear;
-    let monthInViewIndex = monthInViewFromAllMonths();
+    let currentYear = this.currentYear;
```

Comment:
So i guess you're doing all these declarations because you can't access `this.currentYear` wthin the function or callbacks below right? This is valid (and why you often see `var that = this` to get around scoping issues, but its to do with how callbacks work and how the `this` keyword changes meaning normally when referenced inside a callback.

Ive done a bad job at expalining here https://jsfiddle.net/ofnxz3r1/1/

Reply:
hmm, yes, I think it makes sense

Reply:
https://jsfiddle.net/ky04npbq/9/